ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezTask.java

```diff
@@ -515,6 +516,10 @@ DAG build(JobConf conf, TezWork tezWork, Path scratchDir, Context ctx,
           Edge e = null;
 
           TezEdgeProperty edgeProp = tezWork.getEdgeProperty(workUnit, v);
+
+          //Add the reducer plan to config to create the combiner object in case of group by.
+          wxConf = GroupByCombiner.setCombinerInConf(v, wxConf, workToConf.get(v));
```

Comment:
Looks like all the combiner creation logic should be a level higher -- would it make sense to have this as part of createVertex method for example? This looks hacky as is

Reply:
During create vertex we may not know the child or parent. The node may get reordered after vertex is created. So to be on safer side it is done here once all the optimizations are done.